File: python/armstrong_numbers.py
def number_to_list(number):
    lst = []
    while number > 0:
        lst.append(number % 10)
        number //= 10
    return lst


def is_armstrong_number(number):
    lst = number_to_list(number)
    p = len(lst)
    return sum(d ** p for d in lst) == number

File: lua/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
    local len = #tostring(number)
    local sum = 0
    local curr = number
    while curr ~= 0 do
        sum = sum + (curr % 10) ^ len
        curr = curr // 10
    end
    return sum == number
end

return ArmstrongNumbers
